src/state.ml

```diff
@@ -354,6 +354,41 @@ let mtu config session =
       in
       tun_mtu - hdr
 
+let control_mtu config state session =
+  (* we assume to have a tun interface and the server send us a tun-mtu *)
+  let tun_mtu =
+    match Config.find Tun_mtu config with
+    | None -> 1500 (* TODO "client_merge_server_config" should do this! *)
+    | Some x -> x
+  and compress = session.compress in
+  match state with
+  | Client_static _ -> assert false
+  | Client_tls_auth _ | Server_tls_auth _ ->
+      (* here, the hash is used from auth *)
+      let hmac = Config.get Auth config |> Mirage_crypto.Hash.digest_size in
+      let not_yet_padded_payload = Packet.id_len + if compress then 1 else 0 in
+      let hdrs =
+        hmac + 1 (* key /op *) + if session.protocol = `Tcp then 2 else 0
```

Comment:
Maybe ``1 (* key / op *) + Packet.hdr_len hmac + if session.protocol = `Tcp then 2 else 0``

Reply:
yes, that's morally what we have in 3938ec1f3f4e5c5822bbdca243970f84ab764b75